tests/test_plugin_loader.py

```
@@ -4463,5 +4594,79 @@ def test_update_existing_repo_to_branch_or_tag_fetch_failure(
+            self.assertEqual(
+                len(caught_exceptions),
+                0,
+                f"Exceptions during concurrent load: {caught_exceptions}",
+            )
+        finally:
+            for name in module_names:
+                sys.modules.pop(name, None)
+
+    def test_lock_is_module_level_and_reentrant_safe(self):
+        """Verify the sys.modules lock exists and is a threading.Lock."""
+        self.assertIsInstance(_SYS_MODULES_LOCK, type(__import__("threading").Lock()))
```

Ruff (0.15.7)

[warning] Use a regular `assert` instead of unittest-style `assertIsInstance`

Replace `assertIsInstance(...)` with `assert ...`

(PT009)